math/SuiteSparse/Portfile

```diff
@@ -18,82 +18,112 @@ long_description            SuiteSparse is a single archive that contains all pa
 
 homepage                    https://people.engr.tamu.edu/davis/suitesparse.html
 
-checksums                   rmd160  c418ad3400283e59ea528fe1d80c1dd9422bb998 \
-                            sha256  8eb2e49e14c0a1a497ed147bc13608e791200e190306ffa50f00be5150fc5306 \
-                            size    65007739
+checksums                   rmd160  8539e35f47729cdc1843e7999b02bca3bc244272 \
+                            sha256  9f976f83405baf55ba24e9494ce3f526251b972f4554ffe9046c21706be87559 \
+                            size    85408692
 
-use_parallel_build          no
-
-patchfiles-append           patch-shared_lib.diff \
-                            patch-Mongoose.diff \
-                            patch-KLU-Include-klu_version.h.diff
+patchfiles-append           patch-KLU-Include-klu_version.h.diff
 
 configure.optflags          -O3
 
-# see https://github.com/macports/macports-ports/pull/2292#issuecomment-408526461
-configure.cppflags-replace  -I${prefix}/include \
-                            -isystem${prefix}/include
+compiler.c_standard         2011
+
+# We keep SUITESPARSE_USE_STRICT at the default value of OFF and 
+# SUITESPARSE_USE_OPENMP at the default value of ON. This way OpenMP will be
+# used only when the compiler supports it. OpenMP is enabled on a case-by-case
+# basis for some subports by adding 'compiler.openmp_version 4.0' and setting
+# PACKAGENAME_USE_OPENMP=ON for CMake.
+#
+# Additionally we set:
+# -DBUILD_STATIC_LIBS=OFF: Disable building static libraries
+# -DSUITESPARSE_INCLUDEDIR_POSTFIX="": Install headers into 'include' instead of 
+# 'include/suitesparse' for compatibility with older packages such as sundials5.
+configure.args-append       -DBUILD_STATIC_LIBS=OFF -DSUITESPARSE_INCLUDEDIR_POSTFIX=""
```

Comment:
Having the headers directly in the `include` folder caused some headaches for macOS users (if they ended up with headers from different version of SuiteSparse installed at the same time).

Please consider installing the headers of SuiteSparse in the new default location and patching SUNDIALS instead.
You could probably use a patch similar to the one applied in MSYS2 for that:
https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-sundials/0004-suitesparse-import-target.patch

Reply:
Alternatively, configure sundials5 with something along the lines of `-DKLU_ENABLE=ON -DKLU_INCLUDE_DIR=$(pkg-config --cflags KLU) -DKLU_LIBRARY_DIR=$(pkg-config.exe --libs-only-L KLU)`.

Reply:
I'm all for it, but won't have the capacity to do this (and test all dependents) in the near future. It would be useful to put this comment in a place where it won't get lost. MacPorts actually has three different sundials versions, and other ports may be broken as well, so doing this right takes time.

> if they ended up with headers from different version of SuiteSparse installed at the same time

MacPorts can only have a single version installed in `/opt/local` though.

Reply:
Yeah. Users ended up with a version from MacPorts, another from Homebrew, another that they built themselves, ...

IIRC, the issue was that they opted for the OpenMP installed by MacPorts (which shared the include folder with some headers from an older version of SuiteSparse) and linked to a newer library that they either built themselves or installed with Homebrew. They ended up with the include directories for both the old and the new version of SuiteSparse in the preprocessor search path. But the one with the old headers happened to be first...
That caused issues that were pretty difficult to debug.

In any case: If the SuiteSparse headers from MacPorts wouldn't have been installed in the same folder as the OpenMP headers, that issue would probably not have occurred like that...

Reply:
I am not against this change, I just don't have the capacity (I'm not the official maintainer of this port). I created an issue so it wouldn't get forgotten: https://trac.macports.org/ticket/69028

Reply:
Macports should not have searched Homebrew prefix (and hopefully vice versa).

Reply:
I'm not sure if Homebrew was involved. The newer SuiteSparse library might have been one built by the user themselves.

Reply:
> I'm not sure if Homebrew was involved. The newer SuiteSparse library might have been one built by the user themselves.

@mmuetzel To be clear, I have nothing against allowing parallel installation of several versions, just someone has to implement that.